src/Compilers/Core/Portable/Compilation/ConversionInfo.cs

```diff
@@ -0,0 +1,147 @@
+
+        public bool Equals(ConversionInfo other)
+            => _kind == other._kind &&
+               EqualityComparer<IMethodSymbol>.Default.Equals(this.MethodSymbol, other.MethodSymbol);
+
+        public override int GetHashCode()
+            => Hash.Combine(this.MethodSymbol, (int)_kind);
+
+        public static bool operator ==(ConversionInfo info1, ConversionInfo info2)
+            => info1.Equals(info2);
+
+        public static bool operator !=(ConversionInfo info1, ConversionInfo info2)
+            => !(info1 == info2);
+
+        private bool HasFlag(ConversionKind kind)
+            => (_kind & kind) == kind;
+
+        /// <summary>
+        /// Returns true if the conversion exists, either as an <see cref="IsWidening"/> or 
+        /// an <see cref="IsNarrowing"/> conversion.
+        /// </summary>
+        public bool Exists => HasFlag(ConversionKind.Exists);
+
+        /// <summary>
+        /// Returns true if the conversion is an identity conversion.
+        /// </summary>
+        public bool IsIdentity => HasFlag(ConversionKind.IsIdentity);
+
+        /// <summary>
+        /// Returns true if the conversion widens from some narrower type to some broader
+        /// type.  For example, a conversion from <see cref="System.String"/> to <see cref="System.Object"/>.
+        /// A widening conversion may generally appear with or without a
+        /// <see cref="IConversionExpression.IsExplicit"/>.  For example, a user in C# could
+        /// write both:
+        /// 
+        /// <code>
+        ///     object o1 = ""; // and
+        ///     object o2 = (object)"";
+        /// </code>
+        /// 
+        /// These would be both be <see cref="IsWidening"/> conversions, but only the second would
+        /// have an <see cref="IConversionExpression.IsExplicit"/>.
+        /// 
+        /// <see cref="IsWidening"/> conversions can appear with other
+        /// conversion types.  For example there are <see cref="IsWidening"/> <see cref="IsReference"/>
+        /// conversions, and <see cref="IsWidening"/> <see cref="IsNumeric"/> conversions.
+        /// </summary>
+        public bool IsWidening => HasFlag(ConversionKind.IsWidening);
+
+        /// <summary>
+        /// True if and only if the conversion narrows from some broader type to some 
+        /// narrower type.  For example, a conversion from <see cref="System.Object"/> to
+        /// <see cref="System.String"/> In general, narrowing conversions appear when there is an
+        /// <see cref="IConversionExpression.IsExplicit"/> present.
+        /// 
+        /// <see cref="IsNarrowing"/> conversions can appear with other
+        /// conversion types.  For example there are <see cref="IsNarrowing"/> <see cref="IsReference"/>
+        /// conversions, and <see cref="IsNarrowing"/> <see cref="IsNumeric"/> conversions.
+        /// </summary>
+        public bool IsNarrowing => HasFlag(ConversionKind.IsNarrowing);
```

Comment:
And yet the summary says
> In general, narrowing conversions appear when there is an`IConversionExpression.IsExplicit` present.

Thus making this inconsistent with `IConversionExpression`.

Can we perhaps have both `IsNarrowing` and `IsExplicit` be synonyms here? Similarly for the widening/implicit?